### DISCUSSION TITLE
Multiple rows cant have the same relations

### DISCUSSION THREAD

**Discussion by dxkyy** [prisma/prisma#15870] [Q&A]
### Bug description

This are my models:
```prisma
model vertretungsplan {
  id         Int       @id @default(autoincrement())
  groups     groups[]
  fach       String
  stunden    Int[]
  vertretung String
  statt      String
  zimmer     String
  info       String
  datum      DateTime
}

model groups{
  id                Int              @id @default(autoincrement())
  jahrgang          Int
  buchstabe         String
  vertretungsplan   vertretungsplan? @relation(fields: [vertretungsplanId], references: [id])
  vertretungsplanId Int?
  user              user[]
}
```

And I am trying to create new entries using express:
```ts
const request = req.body;
  prisma.vertretungsplan
    .create({
      data: {
        klassen: {
          connect: request.klassen.map((c: number) => ({ id: c })),
        },
        fach: request.fach,
        info: request.info,
        statt: request.statt,
        vertretung: request.vertretung,
        zimmer: request.zimmer,
        stunden: request.stunden,
        datum: request.datum,
      },
    })
    .then(async (r) => console.log(r));
```
but it appears that two `vertretungsplan` entries can't have the same `group` entries.
For example when I have a `vertretungsplan` entry that has two `klassen` items:
```js
[
   {id: 1},
   {id: 2}
]
```
but I then try to create a new entry with one `klassen` item of the id `2`, the `vertretungsplan` entry with two `klassen` ids loses the `klassen` item with the id `2`. How do I prevent that?
(if this makes sense)

### How to reproduce

Mentioned in Bug description

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = // my url
}

model vertretungsplan {
  id         Int       @id @default(autoincrement())
  klassen    klassen[]
  fach       String
  stunden    Int[]
  vertretung String
  statt      String
  zimmer     String
  info       String
  datum      DateTime
}

model klassen {
  id                Int              @id @default(autoincrement())
  jahrgang          Int
  buchstabe         String
  vertretungsplan   vertretungsplan? @relation(fields: [vertretungsplanId], references: [id])
  vertretungsplanId Int?
  user              user[]
}

model user {
  id        Int     @id @default(autoincrement())
  username  String
  role      String
  klassen   klassen @relation(fields: [klassenId], references: [id])
  klassenId Int
}

```

```ts
import express, { Express, Request, Response } from "express";
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();
const router = express.Router();

router.post("/", async (req: Request, res: Response) => {
  const request = req.body;
  prisma.vertretungsplan
    .create({
      data: {
        klassen: {
          connect: request.klassen.map((c: number) => ({ id: c })),
        },
        fach: request.fach,
        info: request.info,
        statt: request.statt,
        vertretung: request.vertretung,
        zimmer: request.zimmer,
        stunden: request.stunden,
        datum: request.datum,
      },
    })
    .then(async (r) => console.log(r));
});
```

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: v18.10.0

### Prisma Version

```
prisma                  : 4.3.1
@prisma/client          : 4.5.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine c875e43600dfe042452e0b868f7a48b817b9640b (at ..\..\..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli c875e43600dfe042452e0b868f7a48b817b9640b (at ..\..\..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core c875e43600dfe042452e0b868f7a48b817b9640b (at ..\..\..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt c875e43600dfe042452e0b868f7a48b817b9640b (at ..\..\..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\prisma-fmt-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.3.0-32.c875e43600dfe042452e0b868f7a48b817b9640b
Default Engines Hash    : c875e43600dfe042452e0b868f7a48b817b9640b
Studio                  : 0.473.0
```

**Comment by janpio** [comment]
Hey @dxkyy, your questions seems to have 2 different Prisma schemas (top and bottom one). The top one uses `groups` as does your question, but the bottom one uses `klassen` which your code also seems to be using. That makes it really hard to follow what is going on - can you please fix that?

**Reply by vladi-stevanovic** [reply]
:wave: Hello @dxkyy , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
